src/charm.py

```diff
@@ -64,8 +32,7 @@ def __init__(self, *args: typing.Any):
         self.image_observer = image.Observer(self)
         self.framework.observe(self.on.install, self._on_install)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.build_success, self._on_build_success)
-        self.framework.observe(self.on.build_failed, self._on_build_failed)
```

Comment:
Do we not need this anymore?

Reply:
No because now the build is triggered with `juju-exec` the failure will bubble up to the charm state/log hence it is no longer needed :)

Reply:
How does the charm report new versions of the image? Doesn't it need to trigger the charm for that?